File: service/hospital-service/src/main/java/com/simp/service/hospital/service/HospitalServiceImpl.java
package com.simp.service.hospital.service;

import com.simp.service.shared.domain.service.AuthService;
import com.simp.service.shared.domain.service.HospitalService;
import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public final class HospitalServiceImpl implements HospitalService {
    @NotNull
    private final AuthService authService;

    @NotNull
    public String getGreeting(@NotNull String name) {
        if (name.startsWith("auth-")) {
            return authService.getGreeting(name.substring(5));
        } else {
            return "Hello, %s [from hospital]".formatted(name);
        }
    }
}

File: service/hospital-service/src/main/kotlin/com/simp/service/hospital/service/HospitalServiceImpl.kt
package com.simp.service.hospital.service

import com.simp.service.shared.domain.service.AuthService
import com.simp.service.shared.domain.service.HospitalService
import org.springframework.stereotype.Service

@Service
class HospitalServiceImpl(
    private val authService: AuthService,
) : HospitalService {
    override fun getGreeting(name: String): String = when {
        name.startsWith("auth") -> authService.getGreeting(name.substring(4))
        else -> "Hello, $name [from hospital]"
    }
}